packages/js/data/src/payment-settings/resolvers.ts

```diff
@@ -37,8 +42,21 @@ export function* getPaymentProviders( country?: string ) {
 	}
 }
 
-export function* getOfflinePaymentGateways( country?: string ) {
-	yield getPaymentProviders( country );
+function* getPaymentProvidersIfNeeded( businessCountry?: string ) {
```

Comment:
Ensure each state entry has a resolver that gets the payments providers, if not already fetched.